space-framework/entry.sh

```diff
@@ -22,31 +22,66 @@ function echoerr() {
    echo "ERROR: $1" >&2
 }
 
+function create_or_replace() { # usage: filename
+    filename="$1"
+    kind=$(grep kind: "$filename" | head -1 | awk '{ print $2 }')
+    name=$(grep name: "$filename" | head -1 | awk '{ print $2 }')
+    if kubectl get "$kind" "$name" &> /dev/null ; then
+        kubectl --kubeconfig /home/spacecore/.kube/config replace -f "$filename"
+    else
+        kubectl --kubeconfig /home/spacecore/.kube/config create -f "$filename"
+    fi
+}
+
+function set_config_and_secret_and_crds() {
+    mkdir -p /home/spacecore/.kube
+    KUBECONFIG=/home/spacecore/.kube/config
+    kubectl config set-cluster space-mgt --server="https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}" --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+    kubectl config set-credentials space-mgt --token="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)"
+    kubectl config set-context space-mgt --cluster=space-mgt --user=space-mgt 
+    kubectl config use-context space-mgt
+
+    echo "Delete if it exists and then create a secret for the core cluster."
+    if ! kubectl --kubeconfig /home/spacecore/.kube/config delete secret -n ${NAMESPACE} corecluster ; then
+        echo "Nothing to delete."
+    fi
+    kubectl --kubeconfig /home/spacecore/.kube/config create secret generic -n ${NAMESPACE} corecluster --from-file=kubeconfig=/home/spacecore/.kube/config
+
+    echo "Apply space manager CRDs."
+    for crd in /home/spacecore/config/crds/*.yaml; do
+    create_or_replace $crd
+    done
+}
+
 function run_space_manager() {
     echo "--< Starting space-manager >--"
-    if ! space_manager -v=${VERBOSITY} ; then
-        echoerr "unable to start space-manager!"
-        exit 1
-    fi
+    bin/space-manager --v=${VERBOSITY} --context space-mgt --kubeconfig /home/spacecore/.kube/config &
```

Comment:
@ezrasilvera  I think this is a problem for more than just the space-manager in kubestellar.  There are a few way to solve this.  But to avoid getting hanged on this at this point, I suggest opening an issue and leaving this to post Nov.